src/Microsoft.SqlTools.ServiceLayer/EditData/UpdateManagement/RowCreate.cs

```diff
@@ -227,16 +172,125 @@ public override EditUpdateCellResult SetCell(int columnId, string newValue)
 
         #endregion
 
-        private string GetTableClause()
+        private ScriptBuildResult BuildInsertScript(bool forCommand)
+        {           
+            // Process all the columns in this table
+            List<string> inValues = new List<string>();
+            List<string> inColumns = new List<string>();
+            List<string> outColumns = new List<string>();
+            List<SqlParameter> sqlParameters = new List<SqlParameter>(); 
+            for (int i = 0; i < AssociatedObjectMetadata.Columns.Length; i++)
+            {
+                DbColumnWrapper column = AssociatedResultSet.Columns[i];
+                CellUpdate cell = newCells[i];
+                
+                // Add an out column if we're doing this for a command
+                if (forCommand)
+                {
+                    outColumns.Add($"inserted.{SqlScriptFormatter.FormatIdentifier(column.ColumnName)}");
+                }
+                
+                // Skip columns that cannot be updated
+                if (!column.IsUpdatable)
+                {
+                    continue;
+                }
+                
+                // Make sure a value was provided for the cell 
+                if (cell == null)
+                {
+                    // If there isn't a default, then fail 
+                    if (DefaultValues[i] == null)
+                    {
+                        throw new InvalidOperationException(SR.EditDataCreateScriptMissingValue);
+                    }
+                    
+                    // There is a default value, so trust the db will apply it
+                    continue;
+                }
+
+                // Add the input values
+                if (forCommand)
+                {
+                    // Since this script is for command use, add parameter for the input value to the list
+                    string paramName = $"@Value{RowId}{i}";
```

Comment:
This syntax is making me wonder what is string value and what part is looked up. What'll this be? Will it end up being `@Value12` or something? This is a case where simple string.Format might make it more readable.

Reply:
Also if it is `@Value12` like I'm guessing, could you ever get a conflict between different params? E.g. Row 11 col 1 is `@Value111`, and Row 1 col 11 is also `@Value111`? Shouldn't there be a `_` in the middle to avoid this? Again, maybe i'm missing something...

Reply:
Good flipping catch! I hadn't thought of a scenario like that. At the moment it's not an issue due to only being able to edit one row at a time, but I don't want to be limited to that.